### ISSUE TITLE
RPC Client Hangs

### ISSUE THREAD

**Issue by smouldering** [shadow/shadow-plugin-bitcoin#5]
_Uses Bitcoin 0.9.2 netmine!_

I am trying to perform a simple RPC client command on one of my nodes. This is a step towards me simulating transactions being created and broadcast.

I have a simple topology consisting of one node (IP: 190.162.111.1). This node has two application elements that run the Bitcoin plugin. 

The first application starts up at 10s and creates a server with an RPC IP of 190.162.111.1 (same as the node) _(arguments -rpcbind=190.162.111.1 and -rpcport=8332)_. I had to put in a little code to accept the _rpcbind_ argument which, uses the same code but to create an IPv4 address of the that IP rather than defaulting onto the ANY IP.

The second application starts at 15s and creates an RPC client with arguments _-rpcconnect=190.162.111.1_ and _-rpcport=8332_.

When I scan the log and see everything being setup and have some extra logging to show if the server receives any connection, I notice the client making a call to _d.connect(...)_ and nothing after that (doesn't proceed onto the next line of code, rpcclient.cpp line 52).

The rest of a the log is full of heartbeats. So, has RPC client -> server been tested before in Shadow? What were the arguments used? Was it just a local connection on 127.0.0.1 -> 0.0.0.0?

**Comment by amiller** [comment]
I do not believe we have tested this successfully with any RPC interaction. I'll see if I can find any notes on why not.

**Comment by smouldering** [comment]
Ok, yeah if you could find some notes or perhaps a rough reason as to why that would be useful for me. Thanks!
